engines/ags/engine/main/config.cpp

```diff
@@ -315,10 +336,31 @@ void apply_config(const ConfigTree &cfg) {
 
 		// Translation / localization
 		Common::String translation;
-		if (ConfMan.getActiveDomain()->tryGetVal("translation", translation) && !translation.empty())
-			_GP(usetup).translation = translation;
-		else
-			_GP(usetup).translation = CfgReadString(cfg, "language", "translation");
+		const Common::String config_language = ConfMan.get("language");
+		if (!config_language.empty()) {
```

Comment:
I think if it might be a good idea here to add a check on `ConfMan.isKeyTemporary("language")`, meaning that the language comes from the command line.

We do have a `language` field for some AGS games in the config file that corresponds to the main language of the game (so not to a translation file). Normally we only do that for mono-lingual games that have no translations, but with the number of games we have in the detection table it is easy to overlook some games. And normally you would not have a translation file matching the language code. So it should still pick up the `translation` value if there is one. But there might be some edge cases, such as a game tagged as french in the detection (so with `language=fr` in the config file), and a translation file `french_AD.tra` for an alternative French text. With such an edge case and your current code it would be impossible for the user to select and use a different language in the GUI.

Also, and more importantly, if you make the change I suggested above to allow selecting English for games with a different main language that have an English translation, your code as it is currently would not allow playing any other language selected in the GUI (because [we register a default of "en" for the language](https://github.com/scummvm/scummvm/blob/b073fda7e25b2df000fd133a4214f289b36a62ca/base/commandLine.cpp#L334)).